### PR TITLE
Add non-normative section about the DID Method Registry, re: #133

### PR DESCRIPTION
**Pull request by rhiaro**
A starting point; I expect people will have lots of opinions about what goes in here.

The **Unique DID Method Names** section tries to strike a balance between explaining why the DID Method Registry is useful, without implying it's authoritative or required, and without being too defensive about its existence at all.

@talltree's original proposal was to call it "Governance of DID Method Namespaces" but I thought "governance" implied a bit more authority or even just process than perhaps we want to associate with the DID Method Registry. 

Also there might be a better place in the spec for this section.

I also added a *very short* section in Security Considerations because if I understand correctly potential non-unique DID method names could be an attack vector. This might warrant expanding on in the future, but I kept it brief for now.

***
<a href="https://pr-preview.s3.amazonaws.com/w3c-ccg/did-spec/pull/186.html" title="Last updated on Jun 3, 2019, 6:51 AM UTC (48070d9)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c-ccg/did-spec/186/f892d6c...48070d9.html" title="Last updated on Jun 3, 2019, 6:51 AM UTC (48070d9)">Diff</a>

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -2049,6 +2038,38 @@ <h3>
...
+expected to use a method name that is unique among all DID method
+names known to them at the time of publication.
+      </p>
+      <p>
+Because there is no central authority for allocating or approving DID
+methods, there is no way to know for certain if a particular DID method
+name is unique. To help with this challenge, the
+<a href="https://www.w3.org/community/credentials/">W3C Credentials Community
+Group</a> maintains a non-authoritative list of known DID
+method names and their associated specifications (see Appendix
```
Comment (yancyribbens) [inline index.html]: I like the explanation of why did method registry is useful but not a point of centralization 👍

File: index.html [hunk 2]
```diff
@@ -2049,6 +2038,38 @@ <h3>
...
+<a href="#registries"></a>).
+      </p>
+      <p>
+The [[DID-METHOD-REGISTRY]] is a tool for implementors to use when coming
```
Comment (yancyribbens) [inline index.html]: The w3c DID-METHOD-REGISTRY?  Is it worth noting here that we are referring to a specific instance (the w3c registry) here?

**Comment by peacekeeper** [comment]
Per discussion in [2019-05-09 DID Spec and DID Resolution Spec meeting](https://github.com/w3c-ccg/meetings/tree/gh-pages/2019-05-09-did-spec-and-resolution), @jandrieu will review and potentially propose changes.

**Comment by peacekeeper** [comment]
Ping @jandrieu: We discussed this on last Thursday's [call](https://docs.google.com/document/d/1qYBaXQMUoB86Alquu7WBtWOxsS8SMhp1fioYKEGCabE/); do you have any further feedback on this? Otherwise I think we'll merge it soon.

**Comment by peacekeeper** [comment]
Per discussion in [2019-05-30 DID Spec and DID Resolution Spec](https://github.com/w3c-ccg/meetings/tree/gh-pages/2019-05-30-did-spec-and-resolution) meeting, this will be merged in 3 days from now unless there is any additional feedback.

**Pull request merged**